parsing/intermediate/dialects/dialect1/parserImpls.ftl

```diff
@@ -963,11 +964,166 @@ SqlCreate SqlCreateTable() :
             }
         }
     ]
+    [
+        <PARTITION> <BY>
+        partition = CreateTablePartitionBy()
+    ]
     onCommitType = OnCommitTypeOpt()
     {
         return new SqlCreateTableDialect1(s.end(this), createSpecifier, setType,
          volatility, ifNotExists, id, tableAttributes, columnList, query,
-         withData, primaryIndex, indices, onCommitType);
+         withData, primaryIndex, indices, partition, onCommitType);
+    }
+}
+
+SqlTablePartition CreateTablePartitionBy() :
+{
+    final SqlNodeList partitions = new SqlNodeList(getPos());
+    SqlNode e;
+}
+{
+    (
+        e = PartitionExpression()
+        { partitions.add(e); }
+    |
+        <LPAREN>
+        e = PartitionExpression()
+        { partitions.add(e); }
+        (
+            e = PartitionExpression()
+            { partitions.add(e); }
+        )*
+        <RPAREN>
+    )
+    { return new SqlTablePartition(getPos(), partitions); }
+}
+
+SqlNode  PartitionExpression() :
+{
+    final SqlNode e;
+    int constant = 0;
+}
+{
+    (
+        e = RangeN()
+    |
+        e = CaseN()
+    |
+        e = SqlExtractFromDateTime()
+    |
+        e = PartitionByColumnOption()
+    )
+    [
+        <ADD> { constant = UnsignedIntLiteral(); }
+    ]
+    { return new SqlTablePartitionExpression(getPos(), e, constant); }
+}
+
+SqlNode PartitionByColumnOption() :
+{
+    SqlNode e;
+    final SqlNodeList columnList = new SqlNodeList(getPos());
+}
+{
+    <COLUMN> <ALL> <BUT> <LPAREN>
```

Comment:
nit: wrap lines 1028-1063 in parens and indent

Reply:
Fixed in https://github.com/googleinterns/calcite/pull/219/commits/be29e755ef0eaa3c37c399614f442b507af39d26